code/datums/dna.dm

```diff
@@ -55,7 +55,8 @@ GLOBAL_LIST_INIT(total_uf_len_by_block, populate_total_uf_len_by_block())
 	var/unique_enzymes
 	///Stores the hashed values of traits such as skin tones, hair style, and gender
 	var/unique_identity
-	var/blood_type
+	/// The singleton blood type
+	var/datum/blood_type/blood_type
```

Comment:
Doing it this way means changing species will re-randomize your blood type. Not sure if you care about that but when I handled this I made sure it was maintained.

Reply:
(Particularly sad for clowns. They'll lose their special blood type)